src/Gui/ApplicationPy.cpp

```diff
@@ -1301,37 +1306,61 @@ PyObject* Application::sGetCommandShortcut(PyObject * /*self*/, PyObject *args)
     }
 }
 
+PyObject* Application::sSetCommandShortcut(PyObject * /*self*/, PyObject *args)
+{
+    char* pName;
+    char* pShortcut;
+    if (!PyArg_ParseTuple(args, "ss", &pName, &pShortcut))
+        return NULL;
+
+    Command* cmd = Application::Instance->commandManager().getCommandByName(pName);
+    if (cmd && cmd->getAction()) {
+        Action* action = cmd->getAction();
+        QKeySequence shortcut = QString::fromLatin1(pShortcut);
+        QString nativeText = shortcut.toString(QKeySequence::NativeText);
+        action->setShortcut(nativeText);
+        ParameterGrp::handle hGrp = WindowParameter::getDefaultParameter()->GetGroup("Shortcut");
+        hGrp->SetASCII(pName, pShortcut);
+        return Py::new_reference_to(Py::Boolean(true));
+    }
+    else {
+        PyErr_Format(Base::BaseExceptionFreeCADError, "No such command '%s'", pName);
+        return NULL;
+    }
+}
+
+
 PyObject* Application::sGetCommandInfo(PyObject * /*self*/, PyObject *args)
 {
     char* pName;
     if (!PyArg_ParseTuple(args, "s", &pName))
         return NULL;
-
     Command* cmd = Application::Instance->commandManager().getCommandByName(pName);
-    if (cmd) {
+    if (cmd && cmd->getAction()) {
+        Action* action = cmd->getAction();
         PyObject* pyList = PyList_New(6);
         const char* menuTxt = cmd->getMenuText();
         const char* tooltipTxt = cmd->getToolTipText();
         const char* whatsThisTxt = cmd->getWhatsThis();
         const char* statustipTxt = cmd->getStatusTip();
         const char* pixMapTxt = cmd->getPixmap();
-        const char* shortcutTxt = cmd->getAccel();
-
-#if PY_MAJOR_VERSION >= 3
-        PyObject* strMenuTxt = PyUnicode_FromString(menuTxt ? menuTxt : "");
-        PyObject* strTooltipTxt = PyUnicode_FromString(tooltipTxt ? tooltipTxt : "");
-        PyObject* strWhatsThisTxt = PyUnicode_FromString(whatsThisTxt ? whatsThisTxt : "");
-        PyObject* strStatustipTxt = PyUnicode_FromString(statustipTxt ? statustipTxt : "");
-        PyObject* strPixMapTxt = PyUnicode_FromString(pixMapTxt ? pixMapTxt : "");
-        PyObject* strShortcutTxt = PyUnicode_FromString(shortcutTxt ? shortcutTxt : "");
-#else
-        PyObject* strMenuTxt = PyString_FromString(menuTxt ? menuTxt : "");
-        PyObject* strTooltipTxt = PyString_FromString(tooltipTxt ? tooltipTxt : "");
-        PyObject* strWhatsThisTxt = PyString_FromString(whatsThisTxt ? whatsThisTxt : "");
-        PyObject* strStatustipTxt = PyString_FromString(statustipTxt ? statustipTxt : "");
-        PyObject* strPixMapTxt = PyString_FromString(pixMapTxt ? pixMapTxt : "");
-        PyObject* strShortcutTxt = PyString_FromString(shortcutTxt ? shortcutTxt : "");
-#endif
+        std::string shortcutTxt = action->shortcut().toString().toStdString();
+
+        #if PY_MAJOR_VERSION >= 3

```

Comment:
It seems to me that all pre-processor directives in this project are placed at column 1, so it should probably stay this way.

Reply:
Okay, thanks for reviewing.  I was thinking I had accidentally moved them during editing.  Not sure if people are still building for python 2 anyway at this point.  In retrospect we should have made all the #if's read #if python 2 instead of #if python 3 as that would simplify removing them when it comes time to do so.

Reply:
Well, our test matrix in Travis does build for Python 2; so until we completely deactivate this test, we should still maintain Python 2 compatibility. And about removing Python 2 support, I'm not sure how that will be handled. I guess we will passively ignore Python 2 rather than actively remove Python 2 code.